code/messaging/src/main/java/com/adobe/marketing/mobile/messaging/ContentCardSchemaData.java

```diff
@@ -0,0 +1,143 @@
+/*
+  Copyright 2024 Adobe. All rights reserved.
+  This file is licensed to you under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License. You may obtain a copy
+  of the License at http://www.apache.org/licenses/LICENSE-2.0
+  Unless required by applicable law or agreed to in writing, software distributed under
+  the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+  OF ANY KIND, either express or implied. See the License for the specific language
+  governing permissions and limitations under the License.
+*/
+
+package com.adobe.marketing.mobile.messaging;
+
+import androidx.annotation.Nullable;
+import androidx.annotation.VisibleForTesting;
+import com.adobe.marketing.mobile.MessagingEdgeEventType;
+import com.adobe.marketing.mobile.services.Log;
+import com.adobe.marketing.mobile.util.DataReader;
+import com.adobe.marketing.mobile.util.JSONUtils;
+import java.util.HashMap;
+import java.util.Map;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+// represents the schema data object for a content-card schema
+public class ContentCardSchemaData implements SchemaData {
+    private static final String SELF_TAG = "ContentCardSchemaData";
+    private Object content;
+    private ContentType contentType;
+    private int publishedDate;
+    private int expiryDate;
+    private Map<String, Object> meta;
+
+    PropositionItem parent;
+
+    ContentCardSchemaData(final JSONObject schemaData) {
+        try {
+            this.contentType =
+                    ContentType.fromString(
+                            schemaData.optString(
+                                    MessagingConstants.ConsequenceDetailDataKeys.CONTENT_TYPE));
+            if (contentType.equals(ContentType.APPLICATION_JSON)) {
+                this.content =
+                        JSONUtils.toMap(
+                                schemaData.getJSONObject(
+                                        MessagingConstants.ConsequenceDetailDataKeys.CONTENT));
+            } else {
+                this.content =
+                        schemaData.getString(MessagingConstants.ConsequenceDetailDataKeys.CONTENT);
```

Comment:
Is this a valid use case? Can content ever be a string? If not I think we can remove this else statement

Reply:
content is dictated by content type.  for now, our ui only allows for json values, but the schema itself allows types beyond json